frontend/packages/dev-console/src/components/import/git/GitSection.tsx

```diff
@@ -166,22 +166,38 @@ const GitSection: React.FC<GitSectionProps> = ({
     [builderImages, gitUrlError, setFieldValue],
   );
 
-  const handleDevfileStrategyDetection = React.useCallback(
-    async (gitService: BaseService) => {
-      if (!values.devfile?.devfileSourceUrl) {
-        // No need to check the existence of the file, waste of a call to the gitService for this need
-        const devfileContents = gitService && (await gitService.getDevfileContent());
-        if (!devfileContents) {
-          setFieldValue('devfile.devfileContent', null);
-          setFieldValue('devfile.devfileHasError', true);
-        } else {
-          setFieldValue('devfile.devfileContent', devfileContents);
-          setFieldValue('devfile.devfileHasError', false);
-        }
+  const handleDevfileStrategyDetection = React.useCallback(async () => {
+    const gitService = getGitService(
```

Comment:
Why do you need to create a new git service here? Why not re use the one like before?

Reply:
Because we are updating the devfilePath with what is detected and hence we cannot use the old gitService as it will not have the updated context.